### ISSUE TITLE
Cannot use optional chaining on non-optional value of type 'NSURL'

### ISSUE THREAD

**Issue by konnorbeard** [Moya/Moya#1063]
Similar to #634. 

I'm using Swift 2.3 and Moya 7.0.2, but I'm still seeing this error: 
```
Cannot use optional chaining on non-optional value of type NSURL
```

Here's the offending line: 
```
guard let stringURL = target.baseURL.URLByAppendingPathComponent(target.path)?.absoluteString  else {
```

In #634, it's stated that: "7.0.2 is a Swift 2.3 release (in which ```URLByAppendingPathComponent``` is optional"

So my question: Why am I seeing this error if (1) I have 7.0.2 installed and (2) I'm using Swift 2.3? 

Relevant portions of files: 

Podfile
```
platform :ios, '8.0'
use_frameworks!

target 'XXX' do
    pod 'Moya', '7.0.2'
end
```

Podfile.lock 
```
...
  - Moya (7.0.2):
    - Moya/Core (= 7.0.2)
  - Moya/Core (7.0.2):
    - Alamofire (~> 3.5)
    - Result (~> 2.0)
...
```

Can anyone help with this? Thanks!

**Comment by AndrewSB** [comment]
@konnorbeard: which error? Did you forget to paste in the message?

**Comment by konnorbeard** [comment]
@AndrewSB sorry I wasn't clear - I've updated my original issue to better explain what I'm seeing and what my question is.

**Comment by pedrovereza** [comment]
@konnorbeard Could you try using 7.0.4 and see if the issue still happens? 🤔

**Comment by konnorbeard** [comment]
@pedrovereza just tried, same issue

**Comment by pedrovereza** [comment]
@konnorbeard Hm, that's weird. I know you mentioned you're on Swift 2.3, but can you confirm that 7.0.1 won't work as well because it's targeted as 2.2?

**Comment by pedrovereza** [comment]
@konnorbeard Did you find a fix for this?

**Comment by sunshinejr** [comment]
I'm gonna close this issue due inactivity, feel free to reopen with new info.

**Comment by konnorbeard** [comment]
@pedrovereza @sunshinejr sorry for my inactivity. The errors ended up being swift/xcode version related. The fix ended up being using swift 2.3 with xcode 8.0 - nothing on your guys end. Thanks for your suggestions and time on this.

**Comment by sunshinejr** [comment]
No problem @konnorbeard! Thank you for taking your time to explain what was wrong. It can help someone in the future :)
